andela/phoenix-bn-backend
# ft (emails) - User Can Recieve Email Notification

#### What does this PR do?
- Send email with Verification link to User
- User can click link to verify registration
- Send email to user on successful verification with a redirect link to homepage 

#### How should this be manually tested?
```
Run npm run test for automated test
Run npm run reset:db
Run npm run devstart
```
Go to postman and send a POST request to  the URL below with the request body

```
localhost:3000/api/v1/auth/signup
{
	"firstName":"temi",
	"lastName": "otokuefor",
	"userName": "misa",
	"email": "temipretty1992@gmail.com",
	"password": "password",
	"phoneNumber": "09034343434"
}
```

Click on the `Verify Account` button on the mail to be redirected to the verification endpoint.

```
On success
{
  "status": "success",
  "message": "Account verification was successful"
}
```

```
On error if user has been verified
{
  "status": "error",
  "message": "Unsuccessful, Your account has been Verified."
}
```

```
On error if token has expired
{
  "status": "error",
  "message": "Unauthorized, Link is invalid or expired. Contact page Admin for a new one"
}
```

  
#### Any background context you want to provide?
- Please ensure you use a valid email address 
- You need sendgrid api key, email, home url, app dev url inorder for this feature to work 

#### What are the relevant pivotal tracker stories?
#167765390

#### Implementation Plan
https://docs.google.com/document/d/1Qd6-8Pp4SfmwM7gq19IA12SrQsWvJE5ZBlEqyIubqak/edit?usp=sharing
